src/clj_commons/byte_streams.clj

```diff
@@ -576,35 +577,51 @@
 ;; file => readable-channel
 (def-conversion ^{:cost 0} [File ReadableByteChannel]
   [file]
-  (.getChannel (FileInputStream. file)))
+  (-> file
+      (.toPath)
+      (FileChannel/open (into-array StandardOpenOption
+                                    [StandardOpenOption/READ]))))
 
 ;; file => writable-channel
 (def-conversion ^{:cost 0} [File WritableByteChannel]
   [file {:keys [append?] :or {append? true}}]
-  (.getChannel (FileOutputStream. file (boolean append?))))
+  (let [option-array (into-array StandardOpenOption
+                                 (cond-> [StandardOpenOption/CREATE
+                                          StandardOpenOption/WRITE]
+                                         append?
+                                         (conj StandardOpenOption/APPEND)))]
+    (-> file
+        (.toPath)
+        (FileChannel/open option-array))))
 
 (def-conversion ^{:cost 0} [File (seq-of ByteBuffer)]
   [file {:keys [chunk-size writable?] :or {chunk-size (int 2e9), writable? false}}]
   (let [^RandomAccessFile raf (RandomAccessFile. file (if writable? "rw" "r"))
         ^FileChannel fc (.getChannel raf)
+        close-fn (fn []
+                   (.close raf)
+                   (.close fc))
         buf-seq (fn buf-seq [offset]
-                  (when-not (<= (.size fc) offset)
-                    (let [remaining (- (.size fc) offset)]
+                  (when (and (.isOpen fc)
```

Comment:
It's required, in case the user manually closed the returned closeable-seq. If they close it, that will also .close the RAF and channel, but the seq is still usable, and can be read from up until the point of closing. Attempts to read further would result in Exceptions without checking, this way the seq just ends.